coderd/inboxnotifications.go

```diff
@@ -345,3 +345,34 @@ func (api *API) updateInboxNotificationReadStatus(rw http.ResponseWriter, r *htt
 		UnreadCount:  int(unreadCount),
 	})
 }
+
+// markAllInboxNotificationsAsRead marks as read all unread notifications for authenticated user.
+// @Summary Mark all unread notifications as read
+// @ID mark-all-unread-notifications-as-read
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Notifications
+// @Success 200 {object} codersdk.Response
+// @Router /notifications/inbox/mark-all-as-read [put]
+func (api *API) markAllInboxNotificationsAsRead(rw http.ResponseWriter, r *http.Request) {
+	var (
+		ctx    = r.Context()
+		apikey = httpmw.APIKey(r)
+	)
+
+	err := api.Database.MarkAllInboxNotificationsAsRead(ctx, database.MarkAllInboxNotificationsAsReadParams{
+		UserID: apikey.UserID,
+		ReadAt: sql.NullTime{Time: dbtime.Now(), Valid: true},
+	})
+	if err != nil {
+		api.Logger.Error(ctx, "failed to mark all unread notifications as read", slog.Error(err))
+		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Failed to mark all unread notifications as read.",
+		})
+		return
+	}
+
+	httpapi.Write(ctx, rw, http.StatusOK, codersdk.UpdateInboxNotificationReadStatusResponse{
+		UnreadCount: 0,
+	})
```

Comment:
Actually do we nee to return any HTTP entity? It could be just HTTP 204 meaning "all is read" now.

Reply:
Yeah makes sense too , I'll change it tbh I hesitated between both but agree with it.